### ISSUE TITLE
not working in the authoring tool

### ISSUE THREAD

**Issue by moloko** [cgkineo/adapt-quicknav#40]
I couldn't even get it to upload - 'upload failed' error.

[This post on the forums](https://community.adaptlearning.org/mod/forum/discuss.php?d=2482) reports a slightly different issue in that she could get it to upload but then not actually use it.

**Comment by Newey27** [comment]
Hi, I'm the one who posted in the forum!

As I said in my post: successfully uploaded it, but then on clicking on the component clicks off the menu but doesn't actually add it. I've uploaded a screenshot of the error I got in the console:

https://ibb.co/gLFEa6

I went to have a look, but the code is obfuscated and I couldn't really work anything out.

I'm using the authoring tool v. 0.4.0 and 3.0.0 of the quicknav plugin.

**Comment by oliverfoster** [comment]
As far as I know @canstudios-louisem has been using it in the AT without trouble?

**Comment by oliverfoster** [comment]
@Newey27 quicknav doesn't use the enum library. I can't see how that bug is related.

**Comment by Newey27** [comment]
Really? That's strange. The error isn't present initially but then pops up immediately after I try to select the quick-nav component, so I just assumed it was related. But even if that error isn't from the quicknav, I still can't actually use the component in my courses.

**Comment by oliverfoster** [comment]
Ahh, it's an AT user interface bug you've sent, that makes a little more sense.
That would suggest there is an issue with the schema in that case.

**Comment by Newey27** [comment]
Ah okay. Is there anything in particular I should look for? I've had some problems previously with updating and having old code trailing and breaking stuff.

**Comment by oliverfoster** [comment]
Nah, I need someone who knows how to debug it to have a look. A colleague of mine who knows how will be back tomorrow. Just leave it with me.

**Comment by Newey27** [comment]
Oh, phew! Hahahaha!

**Comment by oliverfoster** [comment]
@tomgreenfield could you give me a hand with this when you get a mo?
